pkg/controller/direct/registry/registry.go

```diff
@@ -124,6 +124,9 @@ func IsIAMDirect(groupKind schema.GroupKind) bool {
 	switch groupKind {
 	case schema.GroupKind{Group: "privateca.cnrm.cloud.google.com", Kind: "PrivateCACAPool"}:
 		return true
+	case schema.GroupKind{Group: "run.cnrm.cloud.google.com", Kind: "RunJob"}:
+		// https://cloud.google.com/run/docs/reference/iam/roles
+		return true
```

Comment:
I think this is orthogonal ... I think if you do this you have to implement the IAM support in the direct controller.  But looking at the logs, I see the calls, so you must have implemented it?